modules/core/04-channel/v2/types/expected_keepers.go

```diff
@@ -23,6 +23,4 @@ type ClientKeeper interface {
 	GetClientState(ctx context.Context, clientID string) (exported.ClientState, bool)
 	// GetClientConsensusState gets the stored consensus state from a client at a given height.
 	GetClientConsensusState(ctx context.Context, clientID string, height exported.Height) (exported.ConsensusState, bool)
-	// GetClientCounterparty returns the counterpartyInfo given a clientID
-	GetClientCounterparty(ctx context.Context, clientID string) (clienttypes.CounterpartyInfo, bool)
```

Comment:
I kind of think this should remain but have `clientV2Keeper` implement this interface and only have channelKeeper take in a ClientV2Keeper.

What do you think @gjermundgaraba

Reply:
To me it seems a bit strange to have it on v1, since it is not used there. The only reason to keep it on v1 would be to avoid having to deal with v2. I feel like it would be less indirection and clearer code if we use the v2 keeper directly when needed.

Reply:
@AdityaSripal do you mean to have the ClientV2Keeper basically wrap the client V1 Keeper and extend it with this extra CounterParty functionality?

Reply:
Yes, but i will do this in a separate PR probably after the audit